store/datatree.go

```diff
@@ -29,7 +29,7 @@ func (t dataTree) traverse(bCtx *env.BubblyContext, fn visitFn) (core.DataBlocks
 }
 
 // visitNode is a wrapper around the callback visit function to make sure that
-// all the nodes are visited at least once, and at most once
+// all the Nodes are visited at least once, and at most once
```

Comment:
Have you done a search+replace here and changed these by accident, or is there a reason for all these being capitalised now?

Reply:
I had to export the fields, otherwise json marshalling wouldn't work

Reply:
(and this part must have gotten changed automatically with my refactor)

Reply:
Another approach would be to restrict the scope of change to marshalling methods by using anonymous structs:

[Marshaling structs with private fields](https://riptutorial.com/go/example/14194/marshaling-structs-with-private-fields)

Just an idea, depending on how much do we really want the "node" fields to be private.

Reply:
Neat trick, but I guess we can keep the members public for now.